dfm_tools/meshkernel_helpers.py

```diff
@@ -77,6 +77,43 @@ def meshkernel_delete_withgdf(mk:meshkernel.meshkernel.MeshKernel, coastlines_gd
                          delete_option=meshkernel.DeleteMeshOption(2), #ALL_COMPLETE_FACES/2: Delete all faces of which the complete face is inside the polygon
                          invert_deletion=False)
 
+def meshkernel_delete_withshp(mk:meshkernel.meshkernel.MeshKernel, coastlines_shp, min_area: float=0):
+    """
+    Delete parts of mesh that are inside the shapefile polygon.
+
+    Parameters
+    ----------
+    mk : meshkernel.meshkernel.MeshKernel
+        DESCRIPTION.
+    coastlines_shp : path to the shp file
+        DESCRIPTION.
+
+    Returns
+    -------
+    None.
+
+    ---------------
+    Contributed by Roger Wang (rq.wang@rutgers.edu)
+    
+    """
+    if coastlines_shp == None:
+        print("Need a shape file.")
+        
+    mesh_bnds = mk.mesh2d_get_mesh_boundaries_as_polygons()
+    
+    bbox = (mesh_bnds.x_coordinates.min(), mesh_bnds.y_coordinates.min(), mesh_bnds.x_coordinates.max(), mesh_bnds.y_coordinates.max())
+    
+    coastlines_gdb = gpd.read_file(coastlines_shp, include_fields= ['polygons'], bbox=bbox)
+    
+    for coastline_geom in coastlines_gdb['geometry']:
+        xx, yy = coastline_geom.exterior.coords.xy
+        xx = np.array(xx)
+        yy = np.array(yy)
+        
+        delete_pol_geom = meshkernel.GeometryList(x_coordinates=xx, y_coordinates=yy) #TODO: .copy()/to_numpy() makes the array contiguous in memory, which is necessary for meshkernel.mesh2d_delete()
+        mk.mesh2d_delete(geometry_list=delete_pol_geom, 
+                         delete_option=meshkernel.DeleteMeshOption(2), #ALL_COMPLETE_FACES/2: Delete all faces of which the complete face is inside the polygon
+                         invert_deletion=False)
```

Comment:
This can be replaced with `meshkernel_delete_withgdf()` to avoid duplicate code. This would also align it with `meshkernel_delete_withcoastlines()`. Could you also move your function a bit up in the script, below the similar `meshkernel_delete_withcoastlines()` function would make more sense.